packages/@react-aria/datepicker/src/index.ts

```diff
@@ -14,4 +14,10 @@ export {useDatePicker} from './useDatePicker';
 export {useDateSegment} from './useDateSegment';
 export {useDateField, useTimeField} from './useDateField';
 export {useDateRangePicker} from './useDateRangePicker';
-export * from './useDisplayNames';
+export {useDisplayNames} from './useDisplayNames';
+
+export type {AriaDatePickerProps, AriaDateRangePickerProps} from '@react-types/datepicker';
+export type {AriaDateFieldProps, DateFieldAria} from './useDateField';
+export type {DatePickerAria} from './useDatePicker';
+export type {DateRangePickerAria} from './useDateRangePicker';
+export type {DateSegmentAria} from './useDateSegment';
```

Comment:
New pattern? We aren't currently very consistent about what types we export from our packages. I was thinking we could try to export all input and output types from the hooks explicitly. Note that this also includes some re-exports from `@react-types` as it avoids users needing to install another package separately to use them.